File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        
        ans = nums[0]
        summ = nums[0]
        for i in range(1,len(nums)):
            if nums[i] > nums[i-1]:
                summ += nums[i]
                ans = max(summ,ans)
            else:
                ans = max(summ,ans)
                summ = nums[i]
        return ans

File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.go
func maxAscendingSum(nums []int) int {
    ans := nums[0]
    sum := nums[0]

    for i := 1; i < len(nums); i++ {
        if nums[i] > nums[i-1] {
            sum += nums[i]
            // ans = int(math.Max(float64(ans), float64(sum)))
        }else{
            // ans = int(math.Max(float64(ans), float64(sum)))
            sum = nums[i]
        }
        if sum > ans {
            ans = sum
        }

    }
    return ans
}